pkg/auth/auth.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package auth
+
+import (
+	log "github.com/Sirupsen/logrus"
+	"net/http"
+	"os"
+)
+
+const (
+	usernameEnvVar = "vic_username"
+	passwordEnvVar = "vic_password"
+)
+
+// Authenticator provides a contract for authenticating HTTP handlers
+type Authenticator interface {
+
+	// Authenticate should wrap an http.HandlerFunc with implementation-specific authentication
+	Authenticate(http.HandlerFunc) http.HandlerFunc
+}
+
+// None is an empty struct for disabling authentication
+type None struct{}
+
+// Authenticate when called on None is a noop
+func (n *None) Authenticate(handler http.HandlerFunc) http.HandlerFunc {
+	return handler
+}
+
+// BasicHTTP provides metadata for HTTP Basic Authentication
+type BasicHTTP struct {
+	Username string
```

Comment:
I exported these specifically so they can be overwritten in unit tests w/o making environment variables (the way they're read in when the program is run from the command line) -- thoughts on whether this is ok or if I should follow what I'm about to call "Java Procedure" and make them unexported & write a method to set/modify them?